### PR TITLE
Making docs a bit more visible

### PR DESCRIPTION
**Pull request by Kushmaro**
## Description
Changing index.md to readme.md so it's being auto-displayed by Github.
Also linking from the main README to the docs section.

## Motivation and Context
Making docs more visible.

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +60,8 @@ resource "ec_deployment" "example_minimal" {
 }
 ```
 
+Full resource documentation can be [found here](https://github.com/elastic/terraform-provider-ec/tree/master/docs)
```
Comment (alaudazzi) [inline README.md:63]:
```suggestion
You can find the full resource documentation in the [terraform-provider-ec/docs/](https://github.com/elastic/terraform-provider-ec/tree/master/docs) repo.
```

**Review by alaudazzi — APPROVED** [review]
A minor editing suggestion, otherwise LGTM.

**Review by karencfv — COMMENTED** [review]
Thanks for the PR @Kushmaro! 

According to the terraform provider registry [requirements](https://www.terraform.io/docs/registry/providers/docs.html#format), we need to have the index page named `index.md`  instead of `README.md` for it to show up properly in the registry documentation. Could you set the name of the file back please? :)

**Review by marclop — CHANGES_REQUESTED** [review]
I'd just add the link in the main `README.md` pointing to the `index.md`. If. we wanted any "Index" on the `docs` folder, then we can add a separate `README.md`, but as Karen mentions, the TF provider documentation states that an `index.md` is required to publish the docs in the registry.

**Comment by Kushmaro** [comment]
Thanks @marclop @karencfv , didn't know this was a TF requirement, I've changed the file back as requested

**Review by marclop — APPROVED** [review]

**Review by karencfv — APPROVED** [review]
Thank you!

**Pull request merged**